src/solver/ts-generator/thermal.cpp

```diff
@@ -596,11 +597,83 @@ void GeneratorTempData::operator()(Data::Area& area, Data::ThermalCluster& clust
 }
 } // namespace
 
+
+std::vector<std::pair<std::string, std::string>> splitStringIntoPairs (const std::string& s,
+                                                                       char delimiter1,
+                                                                       char delimiter2)
+{
+    std::vector<std::pair<std::string, std::string>> pairs;
+    std::stringstream ss(s);
+    std::string token;
+
+    while (std::getline(ss, token, delimiter1))
+    {
+        size_t pos = token.find(delimiter2);
+        if (pos != std::string::npos)
+        {
+            std::string first = token.substr(0, pos);
+            std::string second = token.substr(pos + 1);
+            pairs.push_back(std::make_pair(first, second));
+        }
+        else
+        {
+            logs.warning() << "Error with area and cluster: " << token;
+            logs.warning() << "Correct format: \"area1.cluster1;area2.cluster2\"";
+        }
+    }
+
+    return pairs;
+}
+
+std::vector<Data::ThermalCluster*> getClustersToGen(Data::AreaList& areas,
+                                                    const std::string& clustersToGen,
+                                                    bool globalThermalTSgeneration,
+                                                    bool refreshTSonCurrentYear)
+{
+
+    std::vector<Data::ThermalCluster*> clusters;
+
+    if (clustersToGen.empty()) // generate TS for all clusters
```

Comment:
I may miss something, but this function (**getClustersToGen**) should be more simple.
Currently, it knows for which reason a cluster should have its TS generated. It can be because : 
- the user set the clusters input parameters so that : 
  - all thermal clusters TS should be generated
  - for some of them, TS generation was forced (see **gen-ts** parameter)
- the user specified a command line option for some thermal cluster.

This  function should be oblivious to the reason why we generate TS.
Eventually, it should look like : 
```C++
std::vector<Data::ThermalCluster*> getClustersToGen(Data::AreaList& areas,
                                                    bool refreshTSonCurrentYear)
{
    std::vector<Data::ThermalCluster*> clusters;
    areas.each([&clusters, &refreshTSonCurrentYear](Data::Area& area) 
    {
         for (auto cluster : area.thermal.list.all())
             if (cluster->doWeGenerateTS(refreshTSonCurrentYear)
                 clusters.push_back(cluster.get());
    });
    return clusters;
}
```
So, at some point way before this one in the execution (I mean right after input data was read), we should aggregate thermal clusters data from files and from command line.

Reply:
The code looks more readable if we
1. Pick what clusters to run the Generator on
2. Run the generator
We don't want to mix these, in particular we don't want to parse anything in **GenerateThermalTimeSeries**

Reply:
> ok. So what you say it that we don't do that because it's too much work ?

No, this is the intended behavior. Eventually the local flag will disappear (it has no use if we don't do in-memory generation).

Reply:
The separation between cluster selection and generation is a big improvement !

The best target would be to have:
1. the selection logic based on the executable option in the executable code
2. the selection logic based on the study parameters in the simulation code (but we will remove it at some point)

But it does not seem possible with the current implementation where the executable calls the whole simulation function ... 
However in the other PR with a new tool, I would advise to to this separation.